Formula/node.rb

```diff
@@ -56,30 +56,11 @@ def install
 
     if build.with? "npm"
       resource("npm").stage buildpath/"npm_install"
-
-      # make sure npm can find node
-      ENV.prepend_path "PATH", bin
-      # set log level temporarily for npm's `make install`
-      ENV["NPM_CONFIG_LOGLEVEL"] = "verbose"
-      # unset prefix temporarily for npm's `make install`
-      ENV.delete "NPM_CONFIG_PREFIX"
-
-      cd buildpath/"npm_install" do
-        system "./configure", "--prefix=#{libexec}/npm"
```

Comment:
As @DomT4 already said, both npm's `configure`and `make` are just simple wrappers around basic npm commands. `configure` [writes alls arguments passed to it](https://github.com/npm/npm/blob/latest/configure#L32) to a local `npmrc` configuration file (passing in `--prefix=/foo/prefix`will jsut change the default global prefix used by `-g` global installs). `make install` calls just the insource copy of npm with `node bin/npm-cli.js install -g -f`(which reads the prefix from said `npmrc` file).

This might not be that, what you expect from a standard `make install` call, but it's clearly intented by upstream to work that way (strictly mapping make commands to corresponding npm commands).

> Where exactly would upstream expect the original build directory to live when you do ./configure --prefix=/foo/prefix?

The expection is that it's not actual a 'build' directory, but a local clone of the npm repository used for developing npm, which will never gonna be removed, so smylinking will be totaly fine.